### PR TITLE
Always pull sandbox image

### PR DESCRIPTION
**Pull request by SmartManoj**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**
Pull always  for smoother integration

### CODE REVIEW

File: opendevin/runtime/docker/ssh_box.py [hunk 1]
```diff
@@ -731,6 +730,8 @@ def restart_docker_container(self):
 
             # start the container
             logger.info(f'Mounting volumes: {self.volumes}')
+            # update the container image
+            self.docker_client.images.pull(self.container_image)
```
Comment (tobitege) [inline opendevin/runtime/docker/ssh_box.py:734]: This is within `restart_docker_container`, which itself is called in a loop in `__init__`.
Might be cleaner to do the pull outside of the loop, just before the `while n_tries > 0:`?

**Review by xingyaoww — APPROVED** [review]
LGTM!

**Comment by neubig** [comment]
Hmm, does this make startup time much slower? I just pulled the most recent version of OpenDevin and it's stuck at this log step for minutes:
```
20:54:32 - opendevin:INFO: ssh_box.py:249 - Creating new Docker container
```

In the UI it looks like this is stuck for a long time:
```
Initializing agent (may take up to 10 seconds)...
```

This is a major usability issue, so we should possibly consider reverting this?

**Comment by iFurySt** [comment]
I agree with @neubig , we should either force a pull in the Docker command level or during startup. Alternatively, we could pull the image only if it doesn't already exist.

**Comment by neubig** [comment]
I opened an issue here: https://github.com/OpenDevin/OpenDevin/issues/2555

**Pull request merged**